### DISCUSSION TITLE
Changing color theme with dropdown menu

### DISCUSSION THREAD

**Discussion by tomasz-grabarczyk** [TomSchimansky/CustomTkinter#1403] [Q&A] [answered]
Hello there.

First of all – thanks for great work.

Then my question – I wanted to make theme color changer similar to the Appearance Mode from the complex example – but I fail to do so.

I have in the beginning:

`customtkinter.set_default_color_theme("green")`

Then, similar to the appearance, I’d think that it should change the theme color to e.g. “blue”, but it doesn’t do anything:

```
def change_theme_color_event(new_theme_color: str):
customtkinter.set_default_color_theme(new_theme_color)

self.theme_color_option_menu = customtkinter.CTkOptionMenu(self.sidebar_frame,
                                                         values=["blue", "green", "dark-blue"], command=change_theme_color_event)
```

I also tried to directly call
`ThemeManager.load_theme(new_theme_color)`, but it doesn’t work either.

So could you please tell me, what I do wrong here?
Thank you very much in advance.

**Comment by Akascape** [comment, marked as answer]
@tomasz-grabarczyk Theme color cannot be changed like that. 

You have to destroy and define all the widgets again after loading the theme.

Or you can build an option to change the theme which will work after a restart.

**Reply by tomasz-grabarczyk** [reply]
Ok, good to know, thank you very much.

**Reply by avalon60** [reply]
Hi Tomaz. I usually include a preferences option in my apps and store things like theme and mode in a configuration file or a sqlite database  (you could also use json files I suppose). Either way, just read these and set them on app startup. I also save geometry, so that I can restore Windows to the same position as they were left, when I last closed the app.

You can change the appearance mode, without an app restart, it's just the theme change which is limiting.

**Reply by matthew-efseaff** [reply]
Further question:
Similar to `customtkinter.get_appearance_mode()` for light and dark, how does one query the overall color theme? One can set the theme with `customtkinter.set_default_color_theme("blue")`, but I want to `get` it
Something like `customtkinter.get_theme_mode()`, which would result in `green`, `blue`, or `dark-blue` (for the default offering)?

Thank you

-----

Edit:
Searching around in ThemeManager I found this:
customtkinter.ThemeManager._currently_loaded_theme
which returns the names of the default themes mentioned above...

**Reply by Akascape** [reply]
> Searching around in ThemeManager I found this:
customtkinter.ThemeManager._currently_loaded_theme
which returns the names of the default themes mentioned above.

@matthew-efseaff Thats the way to get the current theme name for now. get theme method is not implemented directly.
